python/pyspark/sql/types.py

```diff
@@ -1119,8 +1120,8 @@ def Dict(d):
     class Row(tuple):
 
         """ Row in DataFrame """
-        __DATATYPE__ = dataType
-        __FIELDS__ = tuple(f.name for f in dataType.fields)
+        __datatype = dataType
```

Comment:
It seems that Row will have two reference to dataType, via `_Row__dataType` and `__dataType`, so I'd like to change it to '__dataType__' to avoid double references.